Content.Server/Animals/Components/EggLayerComponent.cs

```diff
@@ -9,44 +10,47 @@ namespace Content.Server.Animals.Components;
 ///     It also grants an action to players who are controlling these entities, allowing them to do it manually.
 /// </summary>
 
-[RegisterComponent]
+[RegisterComponent, Access(typeof(EggLayerSystem)), AutoGenerateComponentPause]
 public sealed partial class EggLayerComponent : Component
 {
-    [DataField]
-    public EntProtoId EggLayAction = "ActionAnimalLayEgg";
+    /// <summary>
+    ///     The item that gets laid/spawned, retrieved from animal prototype.
+    /// </summary>
+    [DataField(required: true)]
+    public List<EntitySpawnEntry> EggSpawn = default!;
```

Comment:
```suggestion
    [DataField(required: true)]
    public List<EntitySpawnEntry> EggSpawn = new();
```

This is just going to throw needlessly if someone uses addcomp.